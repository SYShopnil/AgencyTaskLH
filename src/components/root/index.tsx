import Button from './button';

export { Button };
export { BannerSectionBar } from './banner-bar';
export { SlideCard } from './slide-card/';
export { HeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition';
export { Navbar } from './navbar';
