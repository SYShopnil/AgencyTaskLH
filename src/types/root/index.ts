export type { IButton, INormalButton, IRedirectButton } from './Button/';

export { BtnColorSchema } from './Button/';

export type { IIconStore } from './_icon';
export { IconName as IconEnums } from './_icon';

export type { IBanner, IBannerSectionBar } from './banner-bar/banner.ts';
export type { ISlideCard } from './slide-card';

export type { IHeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition/index';
export type { IFooter } from './footer';

export type {
  ICommonSection,
  IMegaMenu,
  IMegaMenuItems,
  ISubMenuBlock,
  INavbar,
} from './navbar';
