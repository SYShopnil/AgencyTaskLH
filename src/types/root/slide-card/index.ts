export interface ISlideCard {
  cardNo: string;
  heading: string;
  desc: string;
  cardImg: string;
  isActive: boolean;
  slideCardDataTestId? :string
}
