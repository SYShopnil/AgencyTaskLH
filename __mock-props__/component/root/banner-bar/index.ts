import { IBanner, IBannerSectionBar } from '@src/types/root';
import { IconName } from '@src/types/root/_icon';

export const BannerImageMockData: IBanner[] = [
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
];

export const BannerBarSectionMockData: IBannerSectionBar = {
  contents: BannerImageMockData,
  bannerBarSectionTestId: 'mainWrapper',
};
