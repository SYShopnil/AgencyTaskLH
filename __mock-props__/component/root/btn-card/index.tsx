import { CardSize, ColorSchema, IBtnCard } from '@src/types/root';

export const btnCardMockProps: IBtnCard[] = [
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_lg,
    fill: 65,
    data_test_id: 'btnCardOne',
  },
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_sm,
    fill: 50,
    data_test_id: 'btnCardTwo',
  },
];
