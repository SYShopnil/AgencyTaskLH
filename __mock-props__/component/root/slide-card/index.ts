import { ISlideCard } from '@src/types/root';

export const slideCardMockData: ISlideCard = {
  cardImg: '/assets/media/cardImageFour.png',
  cardNo: '02',
  desc: `Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.`,
  heading: 'Pre-processing',
  isActive: true,
  slideCardDataTestId: 'slideCardTestIdOne',
};
