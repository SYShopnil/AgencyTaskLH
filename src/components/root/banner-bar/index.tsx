import { IBannerSectionBar } from '@src/types/root';
import Banner from './Banner';

import React from 'react';

export const BannerSectionBar = ({
  contents: banners,
  bannerBarSectionTestId,
}: IBannerSectionBar) => {
  return (
    <div
      className={`flex justify-center items-center`}
      data-testid={bannerBarSectionTestId || ''}
    >
      {banners.map((banner, index) => {
        return (
          <div
            key={index}
            className={banners.length - 1 != index ? `mr-[3.925rem]` : ''}
          >
            <Banner {...banner} />
          </div>
        );
      })}
    </div>
  );
};

export { Banner };
