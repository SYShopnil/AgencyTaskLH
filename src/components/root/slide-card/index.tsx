import { ISlideCard } from '@src/types/root';
import Image from 'next/image';
import React from 'react';
import ModuleStyleSheet from './slide-card.module.scss';

export const SlideCard = ({
  cardImg,
  cardNo,
  desc,
  heading,
  isActive,
  slideCardDataTestId,
}: ISlideCard) => {
  return (
    <div
      className={`grid grid-cols-12 bg-[#172029] !py-[7.5rem] !px-[5rem] !rounded-[2.5rem]  border-solid border mainWrapper ${
        ModuleStyleSheet['borderStyle']
      } ${isActive && ModuleStyleSheet['active']} ${
        ModuleStyleSheet['mainWrapper']
      } `}
      data-testid={slideCardDataTestId || 'slideCardTestIdOne'}
    >
      {/* left section wrapper */}
      <div className={`col-span-7 mr-[7.75rem]`}>
        <div className={`mb-[1.5rem]`}>
          <p className={`text-[#D3F699] text-3xl`}>{cardNo}</p>
        </div>
        <div className={`mb-[1.5rem]`}>
          <p className={`text-primary_500 text-2xl`}>{heading}</p>
        </div>
        <div>
          <p className={`text-[#FEFEFE] text-lg`}>{desc}</p>
        </div>
      </div>

      {/* right section wrapper */}
      <div className={`col-span-5 flex justify-center items-center`}>
        <div>
          <Image
            src={cardImg}
            alt="Picture of the author"
            width={490}
            height={559}
          />
        </div>
      </div>
    </div>
  );
};
