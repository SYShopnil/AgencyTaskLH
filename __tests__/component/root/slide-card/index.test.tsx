/** TO DO
 *
 *
 *
 */

import { BannerBarSectionMockData } from '@root/__mock-props__/component/root/banner-bar';
import { slideCardMockData } from '@root/__mock-props__/component/root/slide-card';
import { BannerSectionBar, SlideCard } from '@src/components/root';
import { render } from '@testing-library/react';

/** Done
 *
 * Take Snapshot of slide card component
 *
 */

describe('When Slide Card has been rendered', () => {
  it(`Expect Successfully Rendered`, () => {
    const { getByTestId } = render(<SlideCard {...slideCardMockData} />);
    const catchMainWrapperElement = getByTestId(
      slideCardMockData.slideCardDataTestId || 'slideCardTestIdOne'
    );
    expect(catchMainWrapperElement).toBeInTheDocument();
    expect(catchMainWrapperElement).toMatchSnapshot();
  });
});
