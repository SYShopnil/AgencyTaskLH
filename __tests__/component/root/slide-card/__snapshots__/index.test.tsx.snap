// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Slide Card has been rendered Expect Successfully Rendered 1`] = `
<div
  class="grid grid-cols-12 bg-[#172029] !py-[7.5rem] !px-[5rem] !rounded-[2.5rem]  border-solid border mainWrapper borderStyle active mainWrapper "
  data-testid="slideCardTestIdOne"
>
  <div
    class="col-span-7 mr-[7.75rem]"
  >
    <div
      class="mb-[1.5rem]"
    >
      <p
        class="text-[#D3F699] text-3xl"
      >
        02
      </p>
    </div>
    <div
      class="mb-[1.5rem]"
    >
      <p
        class="text-primary_500 text-2xl"
      >
        Pre-processing
      </p>
    </div>
    <div>
      <p
        class="text-[#FEFEFE] text-lg"
      >
        Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.
      </p>
    </div>
  </div>
  <div
    class="col-span-5 flex justify-center items-center"
  >
    <div>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27490%27%20height=%27559%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Picture of the author"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;
